cmd/config.go

```diff
@@ -0,0 +1,217 @@
+	"github.com/ansible/receptor/pkg/services"
+	"github.com/ansible/receptor/pkg/types"
+	"github.com/ansible/receptor/pkg/workceptor"
+	"github.com/ghjm/cmdline"
+	"github.com/spf13/viper"
+)
+
+type Initer interface {
+	Init() error
+}
+
+type Preparer interface {
+	Prepare() error
+}
+
+type Runer interface {
+	Run() error
+}
+
+type ReceptorConfig struct {
+	// Used pointer structs to apply defaults to config
+	Node            *types.NodeCfg
+	Trace           logger.TraceCfg
+	LocalOnly       backends.NullBackendCfg          `mapstructure:"local-only"`
+	LogLevel        *logger.LoglevelCfg              `mapstructure:"log-level"`
+	ControlServices []*controlsvc.CmdlineConfigUnix  `mapstructure:"control-services"`
+	TCPPeers        []*backends.TCPDialerCfg         `mapstructure:"tcp-peers"`
+	UDPPeers        []*backends.UDPDialerCfg         `mapstructure:"udp-peers"`
+	WSPeers         []*backends.WebsocketDialerCfg   `mapstructure:"ws-peers"`
+	TCPListeners    []*backends.TCPListenerCfg       `mapstructure:"tcp-listeners"`
+	UDPListeners    []*backends.UDPListenerCfg       `mapstructure:"udp-listeners"`
+	WSListeners     []*backends.WebsocketListenerCfg `mapstructure:"ws-listeners"`
+	TLSClients      []netceptor.TLSClientConfig      `mapstructure:"tls-clients"`
+	TLSServer       []netceptor.TLSServerConfig      `mapstructure:"tls-servers"`
+	WorkCommands    []workceptor.CommandWorkerCfg    `mapstructure:"work-commands"`
+	WorkKubernetes  []*workceptor.KubeWorkerCfg      `mapstructure:"work-kubernetes"`
+	// WorkPython        []workceptor.WorkPythonCfg       `mapstructure:"work-python"`
+	WorkSigning       workceptor.SigningKeyPrivateCfg  `mapstructure:"work-signing"`
+	WorkVerification  workceptor.VerifyingKeyPublicCfg `mapstructure:"work-verification"`
+	IpRouters         []services.IpRouterCfg           `mapstructure:"ip-routers"`
+	TCPClients        []services.TCPProxyOutboundCfg   `mapstructure:"tcp-clients"`
+	TCPServers        []services.TCPProxyInboundCfg    `mapstructure:"tcp-servers"`
+	UDPClients        []services.TCPProxyInboundCfg    `mapstructure:"udp-clients"`
+	UDPServers        []services.TCPProxyInboundCfg    `mapstructure:"udp-servers"`
+	UnixSocketClients []services.UnixProxyOutboundCfg  `mapstructure:"unix-socket-clients"`
+	UnixSocketServers []services.UnixProxyInboundCfg   `mapstructure:"unix-socket-servers"`
+}
+
+type CertificatesConfig struct {
+	InitCA  certificates.InitCAConfig    `mapstructure:"cert-init"`
+	MakeReq []certificates.MakeReqConfig `mapstructure:"cert-makereq"`
+	SignReq []certificates.SignReqConfig `mapstructure:"cert-signreq"`
+}
+
+func PrintPhaseErrorMessage(configName string, phase string, err error) {
+	fmt.Printf("ERROR: %s for %s on %s phase\n", err, configName, phase)
+}
+
+func ParseConfigs(configFile string) (*ReceptorConfig, *CertificatesConfig, error) {
+	if configFile == "" {
+		fmt.Fprintln(os.Stderr, "Could not locate config file (default is $HOME/.receptor.yaml)")
```

Comment:
Where is the default set? As far as I can see their is no "default" and config needs to not be blank